src/utils/sample/uniform.rs

```diff
@@ -73,14 +73,12 @@ pub(crate) fn sample_uniform_rejection(interval_size: &Z) -> Result<Z, MathError
     }
 
     let bit_size = interval_size.bits() as usize;
-    let bit_vector = sample_bits_uniform(bit_size);
 
-    let random = Z::from(&bit_vector);
-    if &random < interval_size {
-        Ok(random)
-    } else {
-        sample_uniform_rejection(interval_size)
+    let mut random = Z::from(&sample_bits_uniform(bit_size));
+    while &random >= interval_size {
+        random = Z::from(&sample_bits_uniform(bit_size));
```

Comment:
I like the change, but was not able to detect a statistically significant performance improvement.

Reply:
Hmm, ok. I thought so. The compiler probably optimizes the checks s.t. they do not take up any signficant time. Nevertheless, we keep the change, because it definitely does not harm the performance. Correct?

Reply:
yes, we should keep it. I also prefer a loop over recursion, since that makes it more obvious whats going on.